iso/image downloader

smossber opened the pull request:
we will need this for the vm provisioning

jwennerberg commented on roles/rhel-iso-download/tasks/main.yml:
```diff
@@ -0,0 +1,43 @@
+- assert:
+    that:
+    - image_directory is defined
```
var is called `image_destination` in `README.md` and `defaults/main.yml`, but `image_directory` in these tasks

smossber replied:
Ah, changed my mind about the var name in case one wants to use it for downloading qcow images etc. 
Will fix defaults/main.yml and the README!

jwennerberg commented on roles/rhel-iso-download/tasks/main.yml:
```diff
@@ -0,0 +1,43 @@
+- assert:
+    that:
+    - image_directory is defined
+    - image_filename is defined
```
both `image_filename` and `image_file` is referenced. which should it be? `image_file` is in README

jwennerberg commented on adhoc/download_iso.yml:
```diff
@@ -0,0 +1,8 @@
+- hosts: labhost
+  become: true
+  vars:
+    async: true
```
this variable name should be changed. pretty sure `async` is a invalid variable name

smossber replied:
yeah.. lets use something rolespecific_async. 
Noticed that we may want to do this for other roles aswell when we have long running download tasks.

jwennerberg commented on roles/rhel-iso-download/tasks/main.yml:
```diff
@@ -0,0 +1,43 @@
+    dest: "{{ image_directory }}/{{ image_filename }}"
+    mode: 755
+    validate_certs: false
+  when: not async  | default(false)
```
see previous comment on invalid variable name (async)

jwennerberg commented on adhoc/download_iso.yml:
```diff
@@ -0,0 +1,8 @@
+  become: true
+  vars:
+    async: true
+    iso_directory: /var/lib/libvirt/images/
```
not correct variable names for the role. see comments below.

smossber commented:
wow, really botched this PR up!

smossber commented:
```
$ ansible-playbook -i inventory adhoc/download_iso.yml 

PLAY [labhost] ****************************************************************************************************************************************************************************************************

TASK [Gathering Facts] ********************************************************************************************************************************************************************************************
Thursday 24 January 2019  16:34:30 +0100 (0:00:00.061)       0:00:00.061 ****** 
ok: [labhost]

TASK [rhel-iso-download : assert] *********************************************************************************************************************************************************************************
Thursday 24 January 2019  16:34:33 +0100 (0:00:02.638)       0:00:02.699 ****** 
ok: [labhost] => {
    "changed": false, 
    "msg": "All assertions passed"
}

TASK [rhel-iso-download : get certs] ******************************************************************************************************************************************************************************
Thursday 24 January 2019  16:34:33 +0100 (0:00:00.138)       0:00:02.837 ****** 
changed: [labhost]

TASK [rhel-iso-download : set_fact certs] *************************************************************************************************************************************************************************
Thursday 24 January 2019  16:34:34 +0100 (0:00:00.669)       0:00:03.506 ****** 
ok: [labhost]

TASK [rhel-iso-download : get keys] *******************************************************************************************************************************************************************************
Thursday 24 January 2019  16:34:34 +0100 (0:00:00.108)       0:00:03.614 ****** 
changed: [labhost]

TASK [rhel-iso-download : set_fact] *******************************************************************************************************************************************************************************
Thursday 24 January 2019  16:34:34 +0100 (0:00:00.483)       0:00:04.098 ****** 
ok: [labhost]

TASK [rhel-iso-download : download ISO (might take a while)] ******************************************************************************************************************************************************
Thursday 24 January 2019  16:34:35 +0100 (0:00:00.119)       0:00:04.217 ****** 
skipping: [labhost]

TASK [rhel-iso-download : download ISO (fire and forget)] *********************************************************************************************************************************************************
Thursday 24 January 2019  16:34:35 +0100 (0:00:00.084)       0:00:04.302 ****** 
changed: [labhost]

PLAY RECAP ********************************************************************************************************************************************************************************************************
labhost                    : ok=7    changed=3    unreachable=0    failed=0   

Thursday 24 January 2019  16:34:37 +0100 (0:00:02.667)       0:00:06.969 ****** 
=============================================================================== 
rhel-iso-download : download ISO (fire and forget) --------------------------------------------------------------------------------------------------------------------------------------------------------- 2.67s
Gathering Facts -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 2.64s
rhel-iso-download : get certs ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ 0.67s
rhel-iso-download : get keys ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 0.48s
rhel-iso-download : assert --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 0.14s
rhel-iso-download : set_fact ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 0.12s
rhel-iso-download : set_fact certs ------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 0.11s
rhel-iso-download : download ISO (might take a while) ------------------------------------------------------------------------------------------------------------------------------------------------------ 0.08s
```

smossber commented:
```
$ cat adhoc/download_iso.yml 
- hosts: labhost
  become: true
  vars:
    rhel_iso_async: true
    image_dest_dir: /var/lib/libvirt/images/
    image_filename: rhel-server-7.6-x86_64-dvd.iso
  roles:
  - rhel-iso-download
```
```
$ ansible-playbook -i inventory adhoc/download_iso.yml 

PLAY [labhost] ****************************************************************************************************************************************************************************************************

TASK [Gathering Facts] ********************************************************************************************************************************************************************************************
Thursday 24 January 2019  16:34:30 +0100 (0:00:00.061)       0:00:00.061 ****** 
ok: [labhost]

TASK [rhel-iso-download : assert] *********************************************************************************************************************************************************************************
Thursday 24 January 2019  16:34:33 +0100 (0:00:02.638)       0:00:02.699 ****** 
ok: [labhost] => {
    "changed": false, 
    "msg": "All assertions passed"
}

TASK [rhel-iso-download : get certs] ******************************************************************************************************************************************************************************
Thursday 24 January 2019  16:34:33 +0100 (0:00:00.138)       0:00:02.837 ****** 
changed: [labhost]

TASK [rhel-iso-download : set_fact certs] *************************************************************************************************************************************************************************
Thursday 24 January 2019  16:34:34 +0100 (0:00:00.669)       0:00:03.506 ****** 
ok: [labhost]

TASK [rhel-iso-download : get keys] *******************************************************************************************************************************************************************************
Thursday 24 January 2019  16:34:34 +0100 (0:00:00.108)       0:00:03.614 ****** 
changed: [labhost]

TASK [rhel-iso-download : set_fact] *******************************************************************************************************************************************************************************
Thursday 24 January 2019  16:34:34 +0100 (0:00:00.483)       0:00:04.098 ****** 
ok: [labhost]

TASK [rhel-iso-download : download ISO (might take a while)] ******************************************************************************************************************************************************
Thursday 24 January 2019  16:34:35 +0100 (0:00:00.119)       0:00:04.217 ****** 
skipping: [labhost]

TASK [rhel-iso-download : download ISO (fire and forget)] *********************************************************************************************************************************************************
Thursday 24 January 2019  16:34:35 +0100 (0:00:00.084)       0:00:04.302 ****** 
changed: [labhost]

PLAY RECAP ********************************************************************************************************************************************************************************************************
labhost                    : ok=7    changed=3    unreachable=0    failed=0   

Thursday 24 January 2019  16:34:37 +0100 (0:00:02.667)       0:00:06.969 ****** 
=============================================================================== 
rhel-iso-download : download ISO (fire and forget) --------------------------------------------------------------------------------------------------------------------------------------------------------- 2.67s
Gathering Facts -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 2.64s
rhel-iso-download : get certs ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ 0.67s
rhel-iso-download : get keys ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 0.48s
rhel-iso-download : assert --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 0.14s
rhel-iso-download : set_fact ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 0.12s
rhel-iso-download : set_fact certs ------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 0.11s
rhel-iso-download : download ISO (might take a while) ------------------------------------------------------------------------------------------------------------------------------------------------------ 0.08s
```
